src/pages/Product.js

```diff
@@ -0,0 +1,31 @@
+import { getProduct } from "../api/productApi";
+import DetailLayout from "../components/product/detail/DetailLayout";
+import ProductDetailContent from "../components/product/detail/ProductDetailContent";
+import { router } from "../routes";
+
+function Product() {
+  const { productId } = router.getParams({ path: "/product/:productId" });
+
+  const updateProductUI = ({ product }) => {
+    const productDetailContentElement = document.getElementById("product-detail-content");
+    productDetailContentElement.innerHTML = ProductDetailContent({ product, loading: false });
+  };
+
+  const loadProduct = async (productId) => {
+    try {
+      const response = await getProduct(productId);
+
+      updateProductUI({
+        product: response,
+      });
+    } catch (error) {
+      console.error("상품 데이터를 가져오는 중 오류가 발생했습니다:", error);
+    }
+  };
+
+  loadProduct(productId);
```

Comment:
소소) 이렇게 함수가 한번 쓰이는 경우 익명 함수를 써보는 건 어떤가요?

```suggestion
  (async (productId) => {
    try {
      const response = await getProduct(productId);

      updateProductUI({
        product: response,
      });
    } catch (error) {
      console.error("상품 데이터를 가져오는 중 오류가 발생했습니다:", error);
    }
  })(productId);
```

Reply:
catch 에서 console.error 보다 toast 로 UI에 표현해주는 것도 좋을 것 같습니다. 혹은 전역적인 에러바운더리에서 처리할 수 있도록 throw 를 해도 좋을 것 같아유